File: flutter_app/lib/models/medicine_model.dart
class MedicineDetails {
  final String name;
  final String description;
  final String sideEffects;
  final String group;

  MedicineDetails(
      {required this.name,
      required this.description,
      required this.sideEffects,
      required this.group});

  factory MedicineDetails.fromJson(Map<String, dynamic> json) {
    return MedicineDetails(
      name: json['name'],
      description: json['description'],
      sideEffects: json['sideEffects'],
      group: json['group'],
    );
  }
}

class Medicine {
  final String id;
  final String pharmacyId;
  final List<MedicineDetails> medicineDetails;
  final int stockLevel;
  final double price;

  Medicine({
    required this.id,
    required this.pharmacyId,
    required this.medicineDetails,
    required this.stockLevel,
    required this.price,
  });

  factory Medicine.fromJson(Map<String, dynamic> json) {
    return Medicine(
      id: json['_id'], // This is where you get the ID from the JSON object
      pharmacyId: json['pharmacyId'],
      medicineDetails: List<MedicineDetails>.from(
          json['medicineDetails'].map((x) => MedicineDetails.fromJson(x))),
      stockLevel: json['stockLevel'],
      price: json['price'].toDouble(),
    );
  }
}

File: server/models/medicine.model.js
const mongoose = require("mongoose");

const medicineGroupSchema = new mongoose.Schema({
  pharmacyId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Pharmacy",
    required: true,
  },
  name: {
    type: String,
    required: true,
  },
});

const medicineDetailsSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },
    description: {
      type: String,
      required: true,
    },
    sideEffects: {
      type: String,
      required: true,
    },
    group: {
      type: medicineGroupSchema,
      required: true,
    },
  },
  { timestamps: true }
);

const medicineSchema = new mongoose.Schema(
  {
    pharmacyId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Pharmacy",
      required: true,
    },
    medicineDetails: [medicineDetailsSchema],
    stockLevel: {
      type: Number,
      required: true,
    },
    price: {
      type: Number,
      required: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Medicine", medicineSchema);
